### PR TITLE
refactor: change query for index screen

### PR DESCRIPTION
**Pull request by ardasnturk**
- replaced the `Query` component structure in `IndexScreen` with `useQuery` to use the `useFocusEffect` hook
- added the props used in `Query` to `useQuery`
- moved the functions, variables and controls used in the `Query` component above return to fix the problem of the `useIsFocused` hook being run continuously inside the `Query` component every time we open `IndexScreen`
- moved `fetchPolicy` variable above useQuery

SVA-1114

### CODE REVIEW

**Review by d4v3d4vE — APPROVED** [review]
Looks good to me!

File: src/screens/IndexScreen.js
Comment (donni106) [inline src/screens/IndexScreen.js:1]: sadly I have duplicated events in calendar view on a date.

Simulator Screen Shot - iPhone 12 - 2023-09-08 at 15 10 40-tile

I will look deeper into it.

**Review by donni106 — CHANGES_REQUESTED** [review]

File: src/screens/IndexScreen.js
Comment (donni106) [reply src/screens/IndexScreen.js:1]: and after pull to refresh I have some entries of earlier days visible

Simulator Screen Shot - iPhone 12 - 2023-09-08 at 15 14 12

Comment (donni106) [reply src/screens/IndexScreen.js:1]: and sometimes the last entry is duplicated

Simulator Screen Shot - iPhone 12 - 2023-09-08 at 15 22 14-tile

File: src/screens/IndexScreen.js [hunk 1]
```diff
@@ -374,223 +444,150 @@ export const IndexScreen = ({ navigation, route }) => {
...
+            }
+            horizontal={false}
+            sectionByDate={isCalendar ? !showCalendar : true}
+            query={query}
+            queryVariables={queryVariables}
+            fetchMoreData={isConnected ? fetchMoreData : null}
+            refreshControl={
+              <RefreshControl
+                refreshing={refreshing}
+                onRefresh={() => refresh(refetch)}
```
Comment (donni106) [inline src/screens/IndexScreen.js:583]: as `refetch` comes from hook `const { data, loading, fetchMore, refetch } = useQuery(` now, we do not need to pass it to the `refresh` method, at is already defined.

File: src/screens/IndexScreen.js
Comment (donni106) [reply src/screens/IndexScreen.js:1]: refactoring done

**Pull request closed without merging**